### PR TITLE
Add order creation

### PR DESCRIPTION
**Pull request by domenic99**
#4

### CODE REVIEW

File: app/models/client.rb [hunk 1]
```diff
@@ -1,3 +1,11 @@
 # model for clients, this model can makes orders
 class Client < ApplicationRecord
+  # method creates new client if client with this phone doesn't exist
+  def self.current_client(parametrs)
+    client = find_by(phone: parametrs[:phone])
+    client ||= create(parametrs)
+    { 'client_id' => client.id }
+  end
+
+  has_many :orders
```
Comment (EvgeniiK) [inline app/models/client.rb]: better to move it up, under the class declaration

File: app/models/driver.rb [hunk 2]
```diff
@@ -26,4 +26,6 @@ def self.find_first_by_auth_conditions(warden_conditions)
       find_by(phone: conditions[:phone])
     end
   end
+
+  has_many :orders
```
Comment (domenic99) [inline app/models/driver.rb]: Ok

File: app/models/client.rb [hunk 3]
```diff
@@ -1,3 +1,11 @@
 # model for clients, this model can makes orders
 class Client < ApplicationRecord
+  # method creates new client if client with this phone doesn't exist
+  def self.current_client(parametrs)
```
Comment (rakvium) [inline app/models/client.rb]: > `parametrs`
Please use word `params` or `parameters`.

File: app/models/client.rb [hunk 4]
```diff
@@ -1,3 +1,11 @@
...
+    client = find_by(phone: parametrs[:phone])
```
Comment (rakvium) [inline app/models/client.rb]: @domenic99 what about `find_or_initialize_by`?
And what if a user want to change their email, for example?

File: app/controllers/order_controller.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  # here i should create order from params
+  def add_order
+    client = Client.current_client(client_params) # id of client in hash
+    ord = Order.new(client.merge(order_params))
+    if ord.save
+      render json: { 'result' => 'Your order was successfully added. Your id: ' + ord.id.to_s }
+    else
+      render json: { 'error' => 'Something went wrong' }
```
Comment (rakvium) [inline app/controllers/order_controller.rb]: @domenic99, maybe it's better to return `ord.errors`?

Comment (domenic99) [inline app/controllers/order_controller.rb]: good idea )

**Comment by domenic99** [comment]
Also, I am going to work at validation

File: config/routes.rb [hunk 6]
```diff
@@ -1,7 +1,11 @@
 Rails.application.routes.draw do
-  devise_for :drivers # defaults: { format: :json } # из-за undefined method `flash'
+  devise_for :drivers
   post 'auth_user' => 'authentication#authenticate_user'
   get 'home' => 'home#index'
+
+  get 'order' => 'order#index'
```
Comment (rakvium) [inline config/routes.rb]: @domenic99 why don't you use `resources`?

File: app/controllers/order_controller.rb [hunk 7]
```diff
@@ -0,0 +1,26 @@
+class OrderController < ApplicationController
```
Comment (rakvium) [inline app/controllers/order_controller.rb]: @domenic99 why is it `OrderController`, not `OrdersController`? We will have a lot of orders, not a one order :stuck_out_tongue_winking_eye:

Comment (rakvium) [inline app/controllers/order_controller.rb]: @domenic99 as far as you added a controller with a right name, you can now destroy this.

**Comment by rakvium** [comment]
@domenic99 please fix conflicts :wrench:

**Comment by domenic99** [comment]
Ok

**Comment by rakvium** [comment]
@domenic99 what about request specs?

**Comment by domenic99** [comment]
@rakvium Sorry for stupid question. Did you mean tests by saying specs?))

**Comment by rakvium** [comment]
@domenic99 yes. Specs are tests :laughing:

**Comment by domenic99** [comment]
@rakvium Ok. I'll do them in this request.

**Comment by domenic99** [comment]
@rakvium You approved changes 23 hours ago, so I am waiting for your comments for new commits before pushing.

**Comment by rakvium** [comment]
@domenic99 it seems fine enough, you can squash and merge. Please contact with @MiroslavaY and @ivansamoyloff to make sure that you will use the same validations.

**Pull request merged**